app/models/note_tag.rb

```diff
@@ -0,0 +1,20 @@
+# == Schema Information
+#
+# Table name: note_tags
+#
+#  note_id :bigint(8)        not null, primary key
+#  k       :string           default(""), not null, primary key
+#  v       :string           default(""), not null
+#
+# Foreign Keys
+#
+#  note_tags_id_fkey  (note_id => notes.id)
+#
+
+class NoteTag < ApplicationRecord
+  belongs_to :note
+
+  validates :note, :associated => true
+  validates :k, :v, :allow_blank => true, :length => { :maximum => 255 }, :characters => true
```

Comment:
Why are we allowing blank values? Possibly there's an argument for a blank value but certainly not for a key?

Reply:
Blank keys are allowed in other tags. Do we have many of them?

Reply:
125 changesets, 58 relations, 1300 ways (in current versions) but in any case just because we allowed silly things in the past doesn't mean we should continue to allow them in new models.

Reply:
I’m rejecting empty keys in cgimap since many years now. I guess they originate from single object rails code.